### PR TITLE
Add word wrap for short descriptions

### LINKED ISSUE #77652: Very long monospace words cause rustdoc to look strange
I tried this code:

```rust
/// `vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv`
pub struct Item;
```

I expected to see some kind of wrapping in rendered docs, but instead, this adds a horizontal ruler. The ruler allows moving docs 
below left panel:

<details><summary>GIF that shows the problem</summary>
<p>

Peek 2020-10-07 16-49

</p>
</details>

### Meta

`rustdoc 1.48.0-nightly (1fd5b9d51 2020-09-20)`

cc @GuillaumeGomez

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #77652 

Screenshot from 2020-10-08 13-26-18

cc @WaffleLapkin
r? @jyn514

### CODE REVIEW

**Comment by jyn514** [comment]
Will this affect word-wrapping in paragraphs with longer text? I saw there was some undesired behavior in https://github.com/rust-lang/blog.rust-lang.org/pull/508#issuecomment-580916188, will that show up here?

**Comment by jyn514** [comment]
r? @XAMPPRocky 

I don't feel comfortable reviewing this, I don't know enough about CSS.

File: src/librustdoc/html/static/rustdoc.css [hunk 1]
```diff
@@ -390,17 +390,13 @@ nav.sub {
 	cursor: pointer;
 }
 
+.docblock-short {
+	word-wrap: anywhere;
```
Comment (XAMPPRocky) [inline src/librustdoc/html/static/rustdoc.css:394]: `word-wrap` is the non-standard name for that property. It works as an alias in most browsers for `overflow-wrap`.
```suggestion
	overflow-wrap: anywhere;
```

**Comment by XAMPPRocky** [comment]
@jyn514 This should be fine as it mostly just affects really long string of characters, most words won’t be broken up unless you set it to `break-word`. You can see an example in the MDN documentation. https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap

File: src/librustdoc/html/static/rustdoc.css [hunk 2]
```diff
@@ -390,17 +390,13 @@ nav.sub {
...
 }
 
+.docblock-short {
+	overflow-wrap: anywhere;
+}
 .docblock-short p {
 	display: inline;
 }
 
-.docblock-short.nowrap {
```
Comment (jyn514) [inline src/librustdoc/html/static/rustdoc.css:397]: If we're no longer adding styling for this, can we remove the `nowrap` class wherever it's used?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/rustdoc.css:397]: I removed it because we have no `nowrap` class generated anywhere, not the other way around. ;)

**Comment by jyn514** [comment]
@bors r=XAMPPRocky

This seems reasonable; not sure I understand everything it does but it seems better than the status quo.

**Pull request merged**